model/src/test/kotlin/utils/DependencyGraphBuilderTest.kt

```diff
@@ -249,6 +250,38 @@ class DependencyGraphBuilderTest : WordSpec({
                 .build(checkReferences = false)
         }
     }
+
+    "breakCycles()" should {
+        "not break undirected cycles" {
+            val edges = listOf(
+                1 to 2,
+                2 to 3,
+                3 to 4,
+                1 to 4
+            )
+
+            breakCycles(edges) shouldContainExactlyInAnyOrder edges
+        }
+
+        "break a directed cycle with a single node" {
+            val edges = listOf(1 to 1)
+
+            breakCycles(edges) should beEmpty()
+        }
+
+        "break directed cycles involving multiple nodes" {
+            val edges = listOf(
+                1 to 2,
+                2 to 3,
+                3 to 4,
+                4 to 1
+            )
+
+            val result = breakCycles(edges)
+
+            result.intersect(edges) shouldHaveSize 3
```

Comment:
The test is a little bit too light. Shouldn't we test the content of `result` ?

Reply:
The test does test the content.

I disagree with that, because any added strictness would make the test too strict:
The algorithm has the freedom to remove any 1 edge from the circle, and if the algorithm is changed to remove a different one the text shouldn't fail. 

What do you think?

Reply:
Ah yes this is a valid remark.
In this case, testing the amount of edges is the only thing you can do.

Reply:
It's not testing only the amount of edges. The intersection implies that it does test the content.